diff --git a/lib/http2.c b/lib/http2.c
index f06fe26d7..9ef93710c 100644
--- a/lib/http2.c
+++ b/lib/http2.c
@@ -167 +167 @@
-static void h2_stream_hash_free(unsigned int id, void *stream)
+static void h2_stream_hash_free(uint32_t id, void *stream)
diff --git a/lib/vtls/mbedtls.c b/lib/vtls/mbedtls.c
index 6245cc26a..87a0a5af2 100644
--- a/lib/vtls/mbedtls.c
+++ b/lib/vtls/mbedtls.c
@@ -445,6 +445,8 @@ static int mbed_verify_cb(void *ptr, mbedtls_x509_crt *crt,
 {
   struct Curl_cfilter *cf = (struct Curl_cfilter *)ptr;
   struct ssl_primary_config *conn_config = Curl_ssl_cf_get_primary_config(cf);
+  struct ssl_config_data *ssl_config =
+    Curl_ssl_cf_get_config(cf, CF_DATA_CURRENT(cf));
   struct Curl_easy *data = CF_DATA_CURRENT(cf);
 
   if(depth == 0) {
@@ -462,6 +464,8 @@ static int mbed_verify_cb(void *ptr, mbedtls_x509_crt *crt,
   if(!conn_config->verifyhost)
     *flags &= ~MBEDTLS_X509_BADCERT_CN_MISMATCH;
 
+  ssl_config->certverifyresult |= (long)*flags;
+
   if(*flags) {
 #ifndef MBEDTLS_X509_REMOVE_INFO
     char buf[128];
@@ -1034,8 +1038,10 @@ static CURLcode mbed_connect_step1(struct Curl_cfilter *cf,
 {
   struct ssl_connect_data *connssl = cf->ctx;
   struct ssl_primary_config *conn_config = Curl_ssl_cf_get_primary_config(cf);
+  struct ssl_config_data *ssl_config = Curl_ssl_cf_get_config(cf, data);
   CURLcode result;
 
+  ssl_config->certverifyresult = 0;
   if((conn_config->version == CURL_SSLVERSION_SSLv2) ||
      (conn_config->version == CURL_SSLVERSION_SSLv3)) {
     failf(data, "Not supported SSL version");
