diff --git a/library/platform_util.c b/library/platform_util.c
index 35a7eb804..fa29c75a5 100644
--- a/library/platform_util.c
+++ b/library/platform_util.c
@@ -241,0 +242,11 @@
+#elif defined(__PSP__)
+#include <sys/time.h>
+mbedtls_ms_time_t mbedtls_ms_time(void)
+{
+    struct timeval tv;
+    if (gettimeofday(&tv, NULL) != 0) {
+        return time(NULL) * 1000;
+    }
+    return (mbedtls_ms_time_t) tv.tv_sec * 1000
+         + (mbedtls_ms_time_t) tv.tv_usec / 1000;
+}
